search.php

```diff
@@ -1,3 +1,178 @@
+                <path stroke-linecap="round" stroke-linejoin="round" d="M16.5 3.75V16.5L12 14.25 7.5 16.5V3.75m9 0H18A2.25 2.25 0 0 1 20.25 6v12A2.25 2.25 0 0 1 18 20.25H6A2.25 2.25 0 0 1 3.75 18V6A2.25 2.25 0 0 1 6 3.75h1.5m9 0h-9" />
+            </svg>
+        </a>
+        <button id="minimise" class="p-12 hover:text-slate-500 group open">
+            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" class="size-6 group-[.open]:hidden">
+                <path fill-rule="evenodd" d="M2.25 4.5A.75.75 0 0 1 3 3.75h14.25a.75.75 0 0 1 0 1.5H3a.75.75 0 0 1-.75-.75Zm14.47 3.97a.75.75 0 0 1 1.06 0l3.75 3.75a.75.75 0 1 1-1.06 1.06L18 10.81V21a.75.75 0 0 1-1.5 0V10.81l-2.47 2.47a.75.75 0 1 1-1.06-1.06l3.75-3.75ZM2.25 9A.75.75 0 0 1 3 8.25h9.75a.75.75 0 0 1 0 1.5H3A.75.75 0 0 1 2.25 9Zm0 4.5a.75.75 0 0 1 .75-.75h5.25a.75.75 0 0 1 0 1.5H3a.75.75 0 0 1-.75-.75Z" clip-rule="evenodd" />
+            </svg>
+
+            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" class="size-6 group-[:not(.open)]:hidden">
+                <path fill-rule="evenodd" d="M2.25 4.5A.75.75 0 0 1 3 3.75h14.25a.75.75 0 0 1 0 1.5H3a.75.75 0 0 1-.75-.75Zm0 4.5A.75.75 0 0 1 3 8.25h9.75a.75.75 0 0 1 0 1.5H3A.75.75 0 0 1 2.25 9Zm15-.75A.75.75 0 0 1 18 9v10.19l2.47-2.47a.75.75 0 1 1 1.06 1.06l-3.75 3.75a.75.75 0 0 1-1.06 0l-3.75-3.75a.75.75 0 1 1 1.06-1.06l2.47 2.47V9a.75.75 0 0 1 .75-.75Zm-15 5.25a.75.75 0 0 1 .75-.75h9.75a.75.75 0 0 1 0 1.5H3a.75.75 0 0 1-.75-.75Z" clip-rule="evenodd" />
+            </svg>
+        </button>
+    </nav>
+    <main class="group grow h-screen">
+        <section class="group-[.minimised]:h-[calc(100%-6rem)] h-3/4 p-12 overflow-auto">
+            <div class="flex justify-between">
+                <h2 class="text-4xl font-bold mb-6">Search</h2>
+            </div>
+
+            <form class="flex rounded p-3 bg-cyan-950 mb-5">
+                <input type="search" name="search" class="rounded p-2 w-full text-black" />
+                <input type="submit" value="Search" class="ml-3 border border-white rounded p-3 px-5 hover:border-slate-500 hover:text-slate-500 hover:cursor-pointer">
+            </form>
+
+            <div class="grid md:grid-cols-2 lg:grid-cols-3 gap-5 items-start">
+                <?php
+                $output = "";
+                foreach($songSearchResults as $index => $song) {
+                    if ($index == 0) {
+                        $output .= "<div class='rounded p-3 bg-cyan-950'>
+                        <h4 class='mb-3 text-2xl font-bold'>{$song->getArtistName()}</h4>";
+                    } elseif ($song->getArtistName() != $songSearchResults[$index - 1]->getArtistName()) {
+                        $output .= "<div class='rounded p-3 bg-cyan-950'>
+                        <h4 class='mb-3 text-2xl font-bold'>{$song->getArtistName()}</h4>";
+                    }
```

Comment:
This statement (plus the other one for the closing div) is a bit gross (no offense). The fact that each condition outputs the same is a bit dodgy. Really this should just be an if with an or. You might want to make the conditions into varaibles as well to improve readability.

If it read

```
if ($firstSong || $artistIsNew)
```

it would be way nicer

Reply:
If artistIsNew is called before the If statement an error is caused it needs to be in an else if.

Reply:
but some refactoring was done to make it easier to read